### ISSUE TITLE
JavaScriptExecutor only allows one task at a time.

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#2590]
Each instance of JavaScriptExecutor uses a single threaded executor. All deploy, shutdown, preprocessing, postprocessing scripts use the static JavaScriptExecutor in JavaScriptUtil which means all channels that use these will wait on one another.

Also, each FilterTransformer and ResponseTransformer contains their own instance of JavaScriptExecutor, and the implication is that they would not be able to run multiple scripts simultaneously either. We may soon be adding an option to allow a connector to process multiple messages simultaneously, therefore this problem needs to be fixed here as well.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-2658
Reporter: wayneh
Created: 2013-06-18T19:35:23.000-0700

**Comment by rbeckman-nextgen** [comment]
Moved JavaScriptExecutor methods and variables to JavaScriptUtil and made them static.
Removed JavaScriptExecutor and all previous constructed instances. They now reference the method statically. 
Now all executions of JavaScriptTasks go through the one executor in JavaScriptUtil. Connectors, filter and response transformers no longer need to have their own local JavaScriptExecutor.

The executor service now uses a CachedThreadPool so multiple scripts can be run simultaneously for the executor.

Imported Comment. Original Details:
Author: wayneh
Created: 2013-06-19T13:09:53.000-0700

**Comment by rbeckman-nextgen** [comment]
Verified that JavaScript tasks are no longer always sent through the single threaded executor in the old JavaScriptExecutor class.

In Beta 2, I reproduced this by deploying two channels with preprocessors that did a Thread.sleep, and verified that only one of them could run at a time. I then did the same test after this change and verified that multiple preprocessors could run with no problems.

Imported Comment. Original Details:
Author: narupley
Created: 2013-06-27T14:03:06.000-0700
